src/App.jsx

```diff
@@ -1,14 +1,86 @@
+import { useState } from 'react';
 import './App.css';
+import ChatLog from './components/ChatLog';
+import messageData from './data/messages.json';
 
 const App = () => {
+  const [messages, setMessages] = useState(messageData);
+  const [likedCount, setLikedCount] = useState(0);
+  const [remoteColor, setRemoteColor] = useState();
+  const [localColor, setLocalColor] = useState();
+
+  const toggleLikedMessage = (messageId) => {
+    const updatedMessages = messages.map((message) => {
+      return message.id === messageId
+        ? { ...message, liked: !message.liked }
+        : message;
+    });
+
+    const likedList = updatedMessages.filter((message) => message.liked);
+
+    setLikedCount(likedList.length);
```

Comment:
Like mentioned previously, we could ditch the extra piece of state and do the following: 
```javascript
const totalLikes = chatMessages.reduce((count, message) => count + (message.liked ? 1 : 0), 0);

```
Then we could use it like so: 
```javascript
<p className='widget' id='heartWidget'>{totalLikes} ❤️s</p>
```